File: 1231-replace-elements-with-greatest-element-on-right-side/replace-elements-with-greatest-element-on-right-side.cpp
class Solution {
public:
    vector<int> replaceElements(vector<int>& arr) {
        int maxi=-1;
        vector<int> ans(arr.size(),0);
       // ans.push_back(-1);
        for(int i=arr.size()-1;i>=0;i--){
            ans[i]=maxi;
            maxi=max(maxi,arr[i]);
        }
        // reverse(ans.begin(),ans.end());
        return ans;
    }
};

File: 1231-replace-elements-with-greatest-element-on-right-side/replace-elements-with-greatest-element-on-right-side.java
class Solution {
    public int[] replaceElements(int[] arr) {
        if(arr.length == 1){
            return new int[] { -1 };
        }
        int l = arr.length, mx = -1;
        // int[] ans = new int[l];
        for(int i=l-1 ; i>0 ; i--){
            int temp = arr[i];
            arr[i] = mx;
            mx = Math.max(temp, mx);
        }
        arr[0] = mx;
        return arr;
        
    }
}

